ext/standard/basic_functions.stub.php

```diff
@@ -395,3 +395,199 @@ function assert($assertion, $description = null): bool {}
+
+function dirname(string $path, int $levels = 1): string {}
+
+/** @return array|string */
+function pathinfo(string $path, int $options = UNKNOWN) {}
+
+/** @return string|false */
+function stristr(string $haystack, string $needle, bool $part = false) {}
+
+/** @return string|false */
+function strstr(string $haystack, string $needle, bool $part = false) {}
+
+/** @return int|false */
+function strpos(string $haystack, string $needle, int $offset = 0) {}
+
+/** @return int|false */
+function stripos(string $haystack, string $needle, int $offset = 0) {}
+
+/** @return int|false */
+function strrpos(string $haystack, string $needle, int $offset = 0) {}
+
+/** @return int|false */
+function strripos(string $haystack, string $needle, int $offset = 0) {}
+
+/** @return string|false */
+function strrchr(string $haystack, string $needle) {}
+
+function chunk_split(string $str, int $chunklen = 76, string $ending = "\r\n"): string {}
+
+/** @return string|false */
+function substr(string $str, int $start, int $length = UNKNOWN) {}
+
+/**
+ * @param mixed $str
+ * @param mixed $replace
+ * @param mixed $start
+ * @param mixed $length
+ * @return string|array|false
+ */
+function substr_replace($str, $replace, $start, $length = UNKNOWN) {}
+
+function quotemeta(string $str): string {}
+
+function ord(string $character): int {}
+
+function chr(int $codepoint): string {}
+
+function ucfirst(string $str): string {}
+
+function lcfirst(string $str): string {}
+
+function ucwords(string $str, string $delimiters = " \t\r\n\f\v"): string {}
+
+/** @param string|array $from */
+function strtr(string $str, $from, string $to = UNKNOWN): string {}
+
+function strrev(string $str): string {}
+
+function similar_text(string $str1, string $str2, float &$percent = UNKNOWN): int {}
+
+function addcslashes(string $str, string $charlist): string {}
+
+function addslashes(string $str): string {}
+
+function stripcslashes(string $str): string {}
+
+function stripslashes(string $str): string {}
+
+/**
+ * @param string|array $search
+ * @param string|array $replace
+ * @param string|array $subject
+ * @param int $replace_count
+ * @return string|array
+ */
+function str_replace($search, $replace, $subject, &$replace_count = UNKNOWN) {}
+
+/**
+ * @param string|array $search
+ * @param string|array $replace
+ * @param string|array $subject
+ * @param int $replace_count
+ * @return string|array
+ */
+function str_ireplace($search, $replace, $subject, &$replace_count = UNKNOWN) {}
+
+function hebrev(string $str, int $max_chars_per_line = 0): string {}
+
+function hebrevc(string $str, int $max_chars_per_line = 0): string {}
+
+function nl2br(string $str, bool $is_xhtml = true): string {}
+
+/** @param mixed $allowable_tags */
+function strip_tags(string $str, $allowable_tags = UNKNOWN): string {}
+
+/**
+ * @param string|array $locales
+ * @return string|false
+ */
+function setlocale(int $category, $locales, string ...$rest) {}
```

Comment:
```suggestion
function setlocale(int $category, $locales, ...$rest) {}
```

String type is not enforced there.